common/Linux/LnxMisc.cpp

```diff
@@ -70,61 +70,69 @@ std::string GetOSVersionString()
 
 #ifdef DBUS_API
 
-bool ChangeScreenSaverStateDBus(const bool inhibit_requested, const char* program_name, const char* reason)
+static bool SetScreensaverInhibitDBus(const bool inhibit_requested, const char* program_name, const char* reason)
 {
 	static dbus_uint32_t s_cookie;
-	// "error_dbus" doesn't need to be cleared in the end with "dbus_message_unref" at least if there is
-	// no error set, since calling "dbus_error_free" reinitializes it like "dbus_error_init" after freeing.
+	const char* bus_method = (inhibit_requested) ? "Inhibit" : "UnInhibit";
 	DBusError error_dbus;
-	dbus_error_init(&error_dbus);
 	DBusConnection* connection = nullptr;
+	static DBusConnection* s_comparison_connection;
 	DBusMessage* message = nullptr;
 	DBusMessage* response = nullptr;
-	// Initialized here because initializations should be before "goto" statements.
-	const char* bus_method = (inhibit_requested) ? "Inhibit" : "UnInhibit";
-	// "dbus_bus_get" gets a pointer to the same connection in libdbus, if exists, without creating a new connection.
-	// this doesn't need to be deleted, except if there's an error then calling "dbus_connection_unref", to free it,
-	// might be better so a new connection is established on the next try.
-	if (!(connection = dbus_bus_get(DBUS_BUS_SESSION, &error_dbus)) || (dbus_error_is_set(&error_dbus)))
-		goto cleanup;
-	if (!(message = dbus_message_new_method_call("org.freedesktop.ScreenSaver", "/org/freedesktop/ScreenSaver", "org.freedesktop.ScreenSaver", bus_method)))
-		goto cleanup;
-	// Initialize an append iterator for the message, gets freed with the message.
 	DBusMessageIter message_itr;
+
+	dbus_error_init(&error_dbus);
+	// Calling dbus_bus_get() after the first time returns a pointer to the existing connection.
+	connection = dbus_bus_get(DBUS_BUS_SESSION, &error_dbus);
+	if (!connection || (dbus_error_is_set(&error_dbus)))
+		goto error_cleanup;
```

Comment:
Try to avoid `goto`. There's some cases where it makes sense, but it causes issues if there's any non-POD types constructed inbetween.

That type of error cleanup makes sense in C, but in C++, we have `ScopedGuard` for this sort of thing (you can cancel so it doesn't destroy on success). See VMManager init for an example. Or, unique_ptr with a custom deleter.